robot_calibration/src/sensor_data_generator.cpp

```diff
@@ -29,88 +33,98 @@ class Listener
     {
       ros::shutdown();
     }
-    
-    //link lengths hardcoded
+
+    //begin sequence to store all data in yaml file
+    yaml_out_stream << YAML::BeginSeq;
+
+    // link lengths hardcoded
     link_1_length_ = 1.8;
     link_2_length_ = 1.0;
 
-    //initial values of theta_1_ and theta_2_ before a joint state message has been received
-    //made to be impossible values so that a "change" in joint state is detected and the angles from the very first message can be written
+    // initial values of theta_1_ and theta_2_ before a joint state message has been received
+    // made to be impossible values so that a "change" in joint state is detected and the angles from the very first
+    // message can be written
     theta_1_ = DBL_MAX;
     theta_2_ = DBL_MAX;
 
-    //initial ee_position calculated based on above joint angle values
+    // initial ee_position calculated based on above joint angle values
     ee_position_[0] = (link_1_length_ * cos(theta_1_)) + (link_2_length_ * cos(theta_1_ + theta_2_));
     ee_position_[1] = (link_1_length_ * sin(theta_1_)) + (link_2_length_ * sin(theta_1_ + theta_2_));
 
-    //setting the angle offsets which are supplied to class as parameters on creation
+    // setting the angle offsets which are supplied to class as parameters on creation
     this->theta_1_offset_ = theta_1_offset;
     this->theta_2_offset_ = theta_2_offset;
   }
 };
 
+//function to take final YAML outstream and write to file
+void Listener::finalWrite()
```

Comment:
Can you add a note saying that a caveat with this approach is that the larger the file becomes, the data is stored in memory before being written to file?